lib/strategy.ts

```diff
@@ -0,0 +1,93 @@
+import { HeaderIndexes, DatasetMapping } from "./models";
```

Comment:
suggestion (non-blocking): strategy.ts is not very clear regarding the intuit of the present file. Imo, a better name would be: sortingStrategies.ts